spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowServerConfiguration.java

```diff
@@ -161,10 +175,97 @@ public void configurePathMatch(PathMatchConfigurer configurer) {
 	}
 
 	@Bean
-	@ConditionalOnMissingBean
+	@ConditionalOnBean(Server.class)
+	public TaskExplorer taskExplorer(DataSource dataSource, Server server) {
```

Comment:
Server dependency here (I have no idea what it is, sorry) is not actually used. Which is weird, as this is a `@ConditionalOnBean(Server.class)`. But this code path ends up doing exactly the same thing as when it is missing (method below)